### PR TITLE
Add phpstan-doctrine

### PR DESCRIPTION
**Pull request by Seldaek**
Blocked by https://github.com/phpstan/phpstan-doctrine/issues/169

### CODE REVIEW

File: tests/object-manager.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+require dirname(__DIR__).'/config/bootstrap.php';
+$kernel = new AppKernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
```
Comment (ondrejmirtes) [inline tests/object-manager.php:4]: There's a typo - needs `App\Kernel` :)

**Comment by Seldaek** [comment]
Thanks again @ondrejmirtes - works fine now except for the `Call to an undefined method Doctrine\Persistence\ObjectManager::getConnection()` nagging me :) There's no way to fix it to see that ManagerRegistry::getManager() returns an EntityManager by any chance?

**Comment by ondrejmirtes** [comment]
@Seldaek Normally you would be able to override it with a custom stub file (https://phpstan.org/user-guide/stub-files) but ManagerRegistry already has a stub in phpstan-doctrine :(

**Comment by Seldaek** [comment]
Ok, not possible to get specialized stubs for EntityManager or the document version for mongo-style ObjectManagers? I don't know enough about internals but I am thinking if it has an instance of the manager via object-manager.php it may be able to load the matching stub corresponding to the concrete type in use in the project?

**Comment by ondrejmirtes** [comment]
Can you point me to the line of code in this repo where PHPStan complains about this?

There's been an issue for supporting multiple EntityManagers through ManagerRegistry for a long time (https://github.com/phpstan/phpstan-doctrine/issues/53) but I don't think we have a solution for that yet. There are some workaround posted as comments... (I'm also not sure it's the same issue you're experiencing.)

**Comment by Seldaek** [comment]
If I up the phpstan level to 2, I get this:

```
  Line   Command/ClearVersionsCommand.php
 ------ -------------------------------------------------------------------------------------------------------
  95     Call to an undefined method Doctrine\Persistence\ObjectManager::getConnection()
```

https://github.com/composer/packagist/blob/main/src/Command/ClearVersionsCommand.php#L95

**Comment by Seldaek** [comment]
The issue you linked is about supporting multiple managers, which is an even more complex setup, but in my case I have just one, and getManager is always going to return an EntityManager and not an ObjectManager

**Pull request merged**